servers/visual/visual_server_raster.cpp

```diff
@@ -7533,10 +7533,10 @@ void VisualServerRaster::_draw_cursors_and_margins() {
 		ERR_CONTINUE( !tex );
 		if (cursors[i].region.has_no_area()) {
 			Point2 size(texture_get_width(tex), texture_get_height(tex));
-			rasterizer->canvas_draw_rect(Rect2(cursors[i].pos, size), 0, Rect2(), tex, Color(1, 1, 1, 1));
+			rasterizer->canvas_draw_rect(Rect2(cursors[i].pos-cursors[i].center, size), 0, Rect2(), tex, Color(1, 1, 1, 1));
```

Comment:
If hotspot is to be a point inside the image, it should be -hotspot, so the
image is moved to the top and left, and the cursor points with that
specific part of the image.

El 23/08/2016 10:15, "Bojidar Marinov" <notifications@github.com> escribió:

> In servers/visual/visual_server_raster.cpp
> <https://github.com/godotengine/godot/pull/6250#discussion_r75819534>:
>
> > @@ -7533,10 +7533,10 @@ void VisualServerRaster::_draw_cursors_and_margins() {
> >  		ERR_CONTINUE( !tex );
> >  		if (cursors[i].region.has_no_area()) {
> >  			Point2 size(texture_get_width(tex), texture_get_height(tex));
> > -			rasterizer->canvas_draw_rect(Rect2(cursors[i].pos, size), 0, Rect2(), tex, Color(1, 1, 1, 1));
> > +			rasterizer->canvas_draw_rect(Rect2(cursors[i].pos-cursors[i].center, size), 0, Rect2(), tex, Color(1, 1, 1, 1));
>
> Hm, should it be a -hotspot or +hotspot?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/godotengine/godot/pull/6250/files/8fcd92c38af19ebfc637e7a0dc623371453d986c#r75819534>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMzBiii69yc_Fsu6lfnr2xtcdBPOmdAJks5qiqwIgaJpZM4JqlD9>
> .
>